src/js/features/metadata/saga.js

```diff
@@ -185,6 +185,15 @@ export function *onSignMessage() {
   yield takeEvery(SIGN_MESSAGE, signMessage);
 }
 
+export function *deleteOffer({offerId}) {
+  const toSend = MetadataStore.methods.removeOffer(offerId);
+  yield doTransaction(DELETE_OFFER_PRE_SUCCESS, DELETE_OFFER_SUCCEEDED, DELETE_OFFER_FAILED, {offerId, toSend});
+}
+
+export function *onDeleteOffer() {
+  yield takeEvery(DELETE_OFFER, deleteOffer);
```

Comment:
```suggestion
  yield takeEvery(DELETE_OFFER, doTransaction.bind(null, DELETE_OFFER_PRE_SUCCESS, DELETE_OFFER_SUCCEEDED, DELETE_OFFER_FAILED));
```
You can just bind doTransaction here and set it up in the action like so:
```
export const deleteOffer = (offerId) => ({
  type: DELETE_OFFER,
  offerId,
  toSend: MetadataStore.methods.removeOffer(offerId)
});
```